uaclient/config.py

```diff
@@ -658,6 +668,26 @@ def help(self, name):
         response_dict["help"] = help_ent.help_info
         return response_dict
 
+    def setup_apt_proxy(self):
```

Comment:
While it makes it easier to reference set.apt_http_proxy ad apt_https_proxy hosting this method on UAConfig, I think it probably makes more sense to locate this function in the `uaclient.apt` module to do everything related to APT from one module.

It may add a bit of complexity to the function signature, but it keep separation of roles more clean. UAConfig should mostly be dealing with sugar/helper methods around parsing contract config (via machine-token reads/writes),  uaclient.conf settings and config notifications & state.  I think side-effects related to the service stacks (APT, livepatch, snap etc) should be housed in modules related to those services instead of growing UAClient utility functions more.

 I'm hoping it's a clean enough function signature to allow for setup_apt_proxy in apt.py to allow two simple params  http_proxy and https_proxy passed  from the config properties which can be called from entitlements.repo.py as you already have below.

What do you think?

Reply:
Definitely! I guess I just didn't think to put it in `uaclient.apt` because I haven't touched that file yet, but that's definitely the place to put this functionality.